apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java

```diff
@@ -90,18 +89,21 @@ public ExtensionInjector extensionInjector() {
     }
 
     /**
-     * Extract the extension context to the given span
+     * Extract the extension context to tracing context
      */
-    void extractExtensionTo(AbstractSpan span) {
-        this.extensionContext.handle(span);
-
+    void extractExtensionTo(TracingContext tracingContext) {
+        tracingContext.getExtensionContext().extract(this);
+        // The extension context could have field not to propagate further, so, must use the this.* to process.
+        this.extensionContext.handle(tracingContext.activeSpan());
```

Comment:
> Seems to break the plugin tests

@kezhenxu94 Yes, the current status of context is complex. `this` and `tracingContext.getExtensionContext()` are not the same thing.

Right now, it should be good.